integration-tests/file-system.test.js

```diff
@@ -8,23 +8,126 @@ import { strict as assert } from 'assert';
 import { test } from 'node:test';
 import { TestRig } from './test-helper.js';
 
-test('reads a file', (t) => {
+test('should be able to read a file', async () => {
   const rig = new TestRig();
-  rig.setup(t.name);
+  await rig.setup('should be able to read a file');
   rig.createFile('test.txt', 'hello world');
 
-  const output = rig.run(`read the file name test.txt`);
+  const result = await rig.run(
+    `read the file test.txt and show me its contents`,
+  );
 
-  assert.ok(output.toLowerCase().includes('hello'));
+  const foundToolCall = await rig.waitForToolCall('read_file');
+
+  // Add debugging information
+  if (!foundToolCall || !result.includes('hello world')) {
+    console.error('Test failed - Debug info:');
+    console.error('Result length:', result.length);
+    console.error('Result (first 500 chars):', result.substring(0, 500));
+    console.error(
+      'Result (last 500 chars):',
+      result.substring(result.length - 500),
+    );
+    console.error('Found tool call:', foundToolCall);
+    console.error('Contains hello world:', result.includes('hello world'));
+
+    // Check what tools were actually called
+    const allTools = rig.readToolLogs();
+    console.error(
+      'All tool calls found:',
+      allTools.map((t) => t.toolRequest.name),
+    );
+  }
```

Comment:
My only nit is i feel like the debug info printing is noisy, makes it harder to parse what the test is actually doing. Super useful, but it feels like something we ought to be able to DRY-up and generalize? Not a blocker though.